symbolic-debuginfo/src/breakpad.rs

```diff
@@ -45,19 +33,30 @@ pub enum BreakpadErrorKind {
     BadEncoding,
 
     /// A record violates the Breakpad symbol syntax.
+    #[deprecated(note = "This is now covered by the Parse variant")]
     BadSyntax,
 
     /// Parsing of a record failed.
+    ///
+    /// The field exists only for API compatibility reasons.
     Parse(&'static str),
+
+    /// The module ID is invalid.
+    InvalidModuleId,
+
+    /// The architecture is invalid.
+    InvalidArchitecture,
 }
 
 impl fmt::Display for BreakpadErrorKind {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             Self::InvalidMagic => write!(f, "missing breakpad symbol header"),
             Self::BadEncoding => write!(f, "bad utf-8 sequence"),
-            Self::BadSyntax => write!(f, "invalid syntax"),
-            Self::Parse(field) => write!(f, "{}", field),
+            Self::Parse(_) => write!(f, "parsing error"),
+            Self::InvalidModuleId => write!(f, "invalid module id"),
+            Self::InvalidArchitecture => write!(f, "invalid architecture"),
+            _ => Ok(()),
```

Comment:
might be good to still list `BadSyntax` explicitly. Note that `#[non_exhaustive]` only forces catch-all branches for external code.